### PR TITLE
Fix #9. compatibility for tostring and tobytes

### LINKED ISSUE #9: Error try to run monitoring
When trying to run following code on Ubuntu MATE 14.04 LTS on intel i7:
```
import gym
env = gym.make("CartPole-v0")
env.monitor.start("/tmp/cartpole-experiment-1")
for i_episode in xrange(20):
    observation = env.reset()
    for t in xrange(100):
        env.render()
        print observation
        action = env.action_space.sample()
        observation, reward, done, info = env.step(action)
        if done:
            print "Episode finished after {} timesteps".format(t+1)
            break

env.monitor.close()
```

I get following error:
```
[2016-04-28 10:59:12,799] Making new env: CartPole-v0
[2016-04-28 10:59:12,800] Creating monitor directory /tmp/cartpole-experiment-1
[2016-04-28 10:59:12,801] Starting new video recorder writing to /tmp/cartpole-experiment-1/openaigym.video.0.8846.video000000.mp4
avconv version 9.18-6:9.18-0ubuntu0.14.04.1, Copyright (c) 2000-2014 the Libav developers
  built on Mar 16 2015 13:19:10 with gcc 4.8 (Ubuntu 4.8.2-19ubuntu1)
Traceback (most recent call last):
  File "temp.py", line 5, in <module>
    observation = env.reset()
  File "/usr/local/lib/python2.7/dist-packages/gym/core.py", line 87, in reset
    self.monitor._after_reset(observation)
  File "/usr/local/lib/python2.7/dist-packages/gym/monitoring/monitor.py", line 237, in _after_reset
    self.video_recorder.capture_frame()
  File "/usr/local/lib/python2.7/dist-packages/gym/monitoring/video_recorder.py", line 111, in capture_frame
    self._encode_image_frame(frame)
  File "/usr/local/lib/python2.7/dist-packages/gym/monitoring/video_recorder.py", line 161, in _encode_image_frame
    self.encoder.capture_frame(frame)
  File "/usr/local/lib/python2.7/dist-packages/gym/monitoring/video_recorder.py", line 284, in capture_frame
    self.proc.stdin.write(frame.tobytes())
AttributeError: 'numpy.ndarray' object has no attribute 'tobytes'
[2016-04-28 10:59:13,196] Finished writing results. You can upload them to the scoreboard via gym.upload('/tmp/cartpole-experiment-1')
```

Clearly the frame returned by `encoder.capture_frame()` is storing returning ndarray in the `frame`. How can I solve this?
PS: Without monitor, code runs perfectly.

### PR DESCRIPTION
**Pull request by carpedm20**

### CODE REVIEW

**Comment by gdb** [comment]
Can we do this without catching the exception? It'd be nicer to know in advance which method we should use, and then just make one attempt. Otherwise we risk catching an unintended error.

**Comment by carpedm20** [comment]
@gdb How about checking the version of numpy? Because `tobytes` was added after `1.9.0` and it will cause error for numpy<=1.8.2.

**Comment by gdb** [comment]
Sounds great!

On Thursday, April 28, 2016, Taehoon Kim <notifications@github.com> wrote:

> @gdb <https://github.com/gdb> How about checking the version of numpy?
> Because tobytes was added after 1.9.0 and it will cause error for
> numpy<=1.8.2.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openai/gym/pull/10#issuecomment-215471322>
>


-- 
Sent from mobile

File: gym/monitoring/video_recorder.py [hunk 1]
```diff
@@ -281,7 +281,10 @@ def capture_frame(self, frame):
         if frame.dtype != np.uint8:
             raise error.InvalidFrame("Your frame has data type {}, but we require uint8 (i.e. RGB values from 0-255).".format(frame.dtype))
 
-        self.proc.stdin.write(frame.tobytes())
+        if np.__version__ >= '1.9.0':
```
Comment (gdb) [inline gym/monitoring/video_recorder.py]: Think we need to do something more sophisticated here:

```
>>> '1.10.0' >= '1.9.0'
False
```

Try using Disutils version parsing, like we do here: https://github.com/openai/mujoco-py/blob/master/mujoco_py/config.py#L45?

Comment (matheusportela) [inline gym/monitoring/video_recorder.py]: This should do the work:

```python
distutils.version.StrictVersion(numpy.__version__) >= distutils.version.StrictVersion('1.9.0')
```

**Comment by carpedm20** [comment]
@gdb: Now the version check is done as:

```python
if distutils.version.StrictVersion(np.__version__) >= distutils.version.StrictVersion('1.9.0'):
    self.proc.stdin.write(frame.tobytes())
else:
    self.proc.stdin.write(frame.tostring())
```

**Comment by gdb** [comment]
Thank you!

**Pull request merged**